DataDog/integrations-core
# [NDM] [Cisco ACI] Fix APIC device status

### What does this PR do?

<img width="1119" alt="image" src="https://github.com/user-attachments/assets/3c4d1a70-5350-4cd4-9533-e109147d5620">

### problem
APIC always report down although they are functionally not (via API, GUI, etc.)


### **proposed solution**
if the device is a controller (the APIC), then utilize the `adSt` field vs. the `fabricSt` which only applies to leaf/spine nodes (as they are a part of the "fabric" vs. the APIC which is the controller / part of monitoring)

**docs referenced**
- https://sandboxapicdc.cisco.com/model-doc/#/objects/fabricNode/properties
- https://pubhub.devnetcloud.com/media/apic-mim-ref-301/docs/MO-fabricNode.html#diagram
`adSt` - The administrative state of the node.
`fabricSt` - Fabric state of node


### **change summary**
* move logic for device status to the response pydantic model
* use new logic for controllers for device status
* update tests

### qa steps
https://datadoghq.atlassian.net/wiki/spaces/II/pages/4013621409/QA+for+Cisco+ACI+integration+public+private+sandbox#Using-ddev
1. `ddev env start --base cisco_aci py3.12`
2. `ddev env config edit cisco_aci py3.12`
```
instances:
  - aci_url: https://sandboxapicdc.cisco.com/
    username: admin
    pwd: "!v3G@!4@Y"
    tls_verify: false
    namespace: zoe-qa-apic
    send_ndm_metadata: true
    tenant:
      - infra
      - Heroes
```
3. `ddev env reload cisco_aci py3.12`
4. `ddev env agent cisco_aci py3.12 check > apic-status.txt`
```
        {
          "device_type": "other",
          "id": "zoe-qa-apic:10.0.0.1",
          "id_tags": [
            "device_ip:10.0.0.1",
            "device_namespace:zoe-qa-apic",
            "device_hostname:apic1",
            "device_id:zoe-qa-apic:10.0.0.1"
          ],
          "integration": "cisco-aci",
          "ip_address": "10.0.0.1",
          "model": "",
          "name": "apic1",
          "serial_number": "TEP-1-1",
          "status": 1,
          "tags": [
            "device_vendor:cisco",
            "source:cisco-aci",
            "apic_role:controller",
            "node_id:1",
            "fabric_state:unknown",
            "fabric_pod_id:1",
            "device_ip:10.0.0.1",
            "device_namespace:zoe-qa-apic",
            "device_hostname:apic1",
            "device_id:zoe-qa-apic:10.0.0.1"
          ],
          "vendor": "cisco",
          "version": "6.0(2h)"
        },
```
status returns as 1 in the metadata, which is available (and true based on the sandbox)

<img width="1426" alt="image" src="https://github.com/user-attachments/assets/b92b2cb6-950f-4b47-9e00-ad5646c572b9">


### Motivation


### Review checklist (to be filled by reviewers)
